convert.go

```diff
@@ -124,6 +131,31 @@ func convertNotEqual(value interface{}) (bson.D, error) {
 	return bson.D{{"$ne", bson.D{{fmt.Sprint(firstArgument), secondArgument}}}}, nil
 }
 
+func convertNot(value interface{}) (bson.D, error) {
+	// if the value is a map, we still need to recursively convert it
+	if isMap(value) {
+		value, _ = internalConvert(value)
+		return bson.D{{"$not", value}}, nil
+	}
+
+	if !isSlice(value) {
+		if isBool(value) {
+			return bson.D{{"$not", value}}, nil
+		}
```

Comment:
@Xelzs  Do you think checking for isBool is enough? Maybe empty strings and numbers should be checked too ?
Meaning :
```suggestion
		if isPrimitive(value) {
			return bson.D{{"$not", value}}, nil
		}
```